pkg/rancher-desktop/utils/paths.ts

```diff
@@ -113,29 +112,24 @@ function getRdctlPath(): string | null {
     basePath = electron.app.isPackaged ? process.resourcesPath : electron.app.getAppPath();
   }
   const osSpecificName = os.platform().startsWith('win') ? `rdctl.exe` : 'rdctl';
-  const rdctlPath = path.join(basePath, 'resources', os.platform(), 'bin', osSpecificName);
 
-  if (!fs.existsSync(rdctlPath)) {
-    return null;
-  }
-
-  return rdctlPath;
+  return path.join(basePath, 'resources', os.platform(), 'bin', osSpecificName);
 }
 
 function getPaths(): Paths {
-  const rdctlPath = getRdctlPath();
   let pathsData: Partial<Paths>;
 
-  if (rdctlPath) {
-    const result = spawnSync(rdctlPath, ['paths'], { encoding: 'utf8' });
+  try {
+    const result = spawnSync(getRdctlPath(), ['paths'], { encoding: 'utf8' });
 
     if (result.status !== 0) {
       throw new Error(`rdctl paths failed: ${ JSON.stringify(result) }`);
```

Comment:
This still doesn't make a lot of sense, because the error gets caught 3 lines later and the payload is dropped.

I think you can do what I suggested previously. Just delete the three lines
```
    if (result.status !== 0) {
      throw new Error(`rdctl paths failed: ${ JSON.stringify(result) }`);
    }
```
and set `pathsData` to that default for any failure.

Reply:
When `spawnSync` calls a command that returns a non-zero exit code, an error is not thrown. You have to check `result.status` (or maybe `result.error`?), so the `if (result.status !== 0) {...}` is needed. Same case for a nonexistent command. See the following code for an example:
```javascript
import child_process from 'child_process';

try {
  const result = child_process.spawnSync('cat', ['doesnotexist.txt'], {encoding: 'utf8'});
  console.log(JSON.stringify(result)); 
} catch (error) {
  console.log(`Got error ${error}`);
}
```